modules/Database/Arangojs.js

```diff
@@ -153,7 +153,9 @@ class ArangoJS {
                              
                             LET trailObjects = (
                                 FILTER startObjects[0] != null
-                                FOR v, e, p IN 0..@depth ANY startObjects[0] ot_edges`;
+                                FOR v, e, p IN 0..@depth ANY startObjects[0] ot_edges
+                                PRUNE LENGTH(p.edges) > 1 && p.edges[-2].relationType == e.relationType
+                                FILTER LENGTH(p.edges) < 1 || p.edges[-2].relationType != p.edges[-1].relationType`;
```

Comment:
We agreed the prune goes inside the condition below, that it's active only when connection types are submitted